drivers/I2C.cpp

```diff
@@ -120,6 +124,82 @@ void I2C::unlock() {
     _mutex->unlock();
 }
 
+int I2C::reset() {
+    lock();
+    // Try i2c_reset() i.e. target specific reset
+    i2c_reset(&_i2c);
+
+    // SCL and SDA as input pins with pullup
+    gpio_init_inout(&_gpio_scl, _scl, PIN_INPUT, PullUp, 1);
+    gpio_init_inout(&_gpio_sda, _sda, PIN_INPUT, PullUp, 1);
+
+    // Read and verify if reset was successful
+    if (gpio_read(&_gpio_scl)) {
+        if (gpio_read(&_gpio_sda)) {
+            // Return successfuly as SDA and SCL is high
+            unlock();
+            // i2c pin initialization required again, since current mux and pin
+            // settings are as GPIO's
+            i2c_init(&_i2c, _sda, _scl);
+            return 0;
+        }
+    } else {
+        // Return as SCL is low and no access to become master.
+        unlock();
+        return I2C_ERROR_BUS_BUSY;
+    }
+
+    // Send clock pulses, for device to recover 2*9
+    int count = 20;
+    while ( (!gpio_read(&_gpio_sda)) && (count > 0)) {
+        count--;
+
+        gpio_mode(&_gpio_scl, PullNone);
```

Comment:
I didn't use DigitalInOut initially because we do not have cleanup functions for gpio's in that, which was missing from hal as well :-(. At present I am using i2c_init and that works fine, so yes I can use DigitalInOut. 
Question: Shouldn't classes be using hal directly and not other wrapper classes which are meant for users? To maintain uniformity and avoid extra layers.